test/server.test.js

```diff
@@ -2742,3 +2742,116 @@ test('should show debug information', function (t) {
 
     t.end();
 });
+
+
+test('should emit \'pre\' event on a 200', function (t) {
+    SERVER.get('/foo/:id', function echoId(req, res, next) {
+        t.ok(req.params);
+        t.equal(req.params.id, 'bar');
+        t.equal(req.isUpload(), false);
+        res.send();
+        next();
+    });
+
+    var count = 0;
+    SERVER.once('pre', function (req, res) {
+        t.ok(req);
+        t.ok(res);
+
+        if (++count === 2) {
```

Comment:
Please remove counts, event emitters are synchronous and there is no race condition here against the pre or route events.